examples/org.eclipse.xtext.xtext.ui.examples/contents/org.eclipse.xtext.example.arithmetics.ui/src/org/eclipse/xtext/example/arithmetics/ui/autoedit/InterpreterAutoEdit.xtend

```diff
@@ -0,0 +1,76 @@
+/** 
+ * Copyright (c) 2010 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.xtext.example.arithmetics.ui.autoedit
+
+import java.math.BigDecimal
+import org.eclipse.jface.text.BadLocationException
+import org.eclipse.jface.text.DocumentCommand
+import org.eclipse.jface.text.IAutoEditStrategy
+import org.eclipse.jface.text.IDocument
+import org.eclipse.xtext.example.arithmetics.arithmetics.Evaluation
+import org.eclipse.xtext.example.arithmetics.arithmetics.Module
+import org.eclipse.xtext.example.arithmetics.arithmetics.Statement
+import org.eclipse.xtext.example.arithmetics.interpreter.Calculator
+import org.eclipse.xtext.nodemodel.ICompositeNode
+import org.eclipse.xtext.nodemodel.util.NodeModelUtils
+import org.eclipse.xtext.resource.XtextResource
+import org.eclipse.xtext.ui.editor.model.IXtextDocument
+
+/** 
+ * @author Sven Efftinge - initial contribution and API
+ * an interactive interpreter as an {@link IAutoEditStrategy}
+ */
+class InterpreterAutoEdit implements IAutoEditStrategy {
```

Comment:
Could be written more idiomatic, e.g. less parentheses, nullsafe method invocation.